### PR TITLE
Use the latest event as the last event in Exercise 4 of Chapter 6

### PR DESCRIPTION
**Pull request by ssmylh**
See #27.

### CODE REVIEW

File: src/main/scala/org/learningconcurrency/exercises/ch6/Ex4.scala [hunk 1]
```diff
@@ -34,57 +34,80 @@ object Ex4 extends App {
...
+    }
 
-    val sScan = s1.scan(2)((s,t)=>s*t)
-    log(s"sScan: element = ${sScan()}")
+    def scan[S](z: S)(f: (S, T) => S): Signal[S] = lastEvent match {
+      case Some(t) => {
+        val s = f(z, t)
+        new Signal(observable.scan(s)(f), s)
+      }
+      case _ => new Signal(observable.scan(z)(f))
```
Comment (axel22) [inline src/main/scala/org/learningconcurrency/exercises/ch6/Ex4.scala]: Here you can use `z` as the initial element of the accumulated signal.

**Comment by axel22** [comment]
Modulo minor remark, looks great! Thanks!

**Comment by ssmylh** [comment]
Thank you for your comment !

Indeed, since the initial element of `Observable.scan(initialValue)(accumulator)` is the `initialValue`,
I fixed so.

**Comment by axel22** [comment]
LGTM!

**Comment by ssmylh** [comment]
I will send other PRs of chapter 6 :)
Thanks.

**Comment by axel22** [comment]
Sounds great - thank you!

**Pull request merged**